components/company/EditForm.vue

```diff
@@ -0,0 +1,370 @@
+<script setup lang="ts">
+  import type { AlertType } from 'composables/useAlerts'
+  const props = defineProps({
+    companyUid: {
+      type: String,
+      required: false,
+      default: null,
+    },
+  })
+  // Fetch event data if props are provided
+  const { hasAccess, user } = useAuthStore()
```

Comment:
Destructuring like this will not preserve reactivity. You can use `storeToRefs(...)` to do this.